### PR TITLE
[RNMobile] - Tooltip for template selection buttons

### LINKED ISSUE #2169: Add a Tooltip for template selection buttons
We need to add a Tooltip to point to the template selection buttons.

### PR DESCRIPTION
**Pull request by geriux**
`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2216
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/14045
`WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/11807

## Description
This PR closes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2169 

Adds a Tooltip to point to the template selection buttons. The component itself has a simple logic and it is attached to the Picker component. If we need another tooltip within the editor we could use this as a base and make it reusable. 

## How has this been tested?

<details><summary>Steps to test</summary>

- Open the main apps PR (iOS/Android).
- Download and install the build.
- Open the app and go to Pages.
- Tap on the add a new page button.
- **Expect** to see the templates picker with the tooltip on top.
- Scroll through the template options.
- **Expect** the tooltip to stay visible.
- Tap on the title of the page.
- **Expect** to see the tooltip fade out.
- Close the app.
- Open the app and go to Pages.
- Tap on the add a new page button.
- **Expect** to see the templates picker **without** the tooltip.

</details>

## Screenshots 

<details><summary>UX Interaction</summary>

iOS | Android
-|-
 | 

</details>

<details><summary>iOS Screenshots</summary>

Tooltip | Tooltip Dark mode
-|-
 | 

</details>

<details><summary>Android Screenshots</summary>

Tooltip | Tooltip Dark mode
-|-
 | 

</details>

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/page-template-picker/picker.native.js [hunk 1]
```diff
@@ -36,13 +46,16 @@ const __experimentalPageTemplatePicker = ( {
 
 	const [ templatePreview, setTemplatePreview ] = useState();
 	const [ pickerVisible, setPickerVisible ] = useState( visible );
+	const [ tooltip, setTooltipVisible ] = useState( false );
```
Comment (mkevins) [inline packages/block-editor/src/components/page-template-picker/picker.native.js:49]: NP: maybe this should be `const [ tooltipVisible, setTooltipVisible ]`, wdyt?

Comment (geriux) [reply packages/block-editor/src/components/page-template-picker/picker.native.js:49]: Oh yes! Thanks for the suggestion, I'll change it =)

Comment (chipsnyder) [reply packages/block-editor/src/components/page-template-picker/picker.native.js:49]: +1 to @mkevins comment here.

Comment (geriux) [reply packages/block-editor/src/components/page-template-picker/picker.native.js]: Updated https://github.com/WordPress/gutenberg/pull/21974/commits/a00bbb6274664ca63bbb4a09a581494e08d1e342

**Review by mkevins — APPROVED** [review]
Tested via steps provided and everything works as expected. LGTM! :+1:

**Pull request merged**